### ISSUE TITLE
Windows PC Don't Need GPU get could not retrieve CUDA device attribute error 

### ISSUE THREAD

**Issue by DavidMulvey2** [tensorflow/tensorflow#38863]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 8
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: n/a
- TensorFlow installed from (source or binary): binary downloaded from tf website
- TensorFlow version: latest from site at 24/04/20 
- Python version: 3.7
- Installed using virtualenv? pip? conda?: pip
- Bazel version (if compiling from source): n/a
- GCC/Compiler version (if compiling from source): n/a
- CUDA/cuDNN version: not applicable
- GPU model and memory: not applicable

Ran Beginners Tensorflow Demo Code in local Python envt (Thonny)

import tensorflow as tf, got 
2020-04-24 09:32:36.144859: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_101.dll'; dlerror: cudart64_101.dll not found
2020-04-24 09:32:36.145180: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.

so as instructed ignored this

then did
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

ran OK
then did
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10)
])
and got the following
2020-04-24 09:32:38.363170: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library nvcuda.dll

2020-04-24 09:32:38.575980: F tensorflow/stream_executor/lib/statusor.cc:34] Attempting to fetch value instead of handling error Internal: Could not retrieve CUDA device attribute (81: UNKNOWN ERROR (1)

So seems need a workaround to get it to continue to ignore the fact there is no GPU (I want it to run on the CPU only.

Thanks very much

**Describe the problem**

**Provide the exact sequence of commands / steps that you executed before running into the problem**

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by oke-aditya** [comment]
Are you able to do a `model.fit(x_train, y_train)`
These warnings occure because you do not have a NVIDIA GPU card or CuDNN installed. It's fine you can continue without it.

**Comment by oke-aditya** [comment]
Tensorflow warns of not having a GPU installed but you can work with CPU as well. To hide these errors and warnings reported by Tensorflow refer [here](https://stackoverflow.com/questions/35911252/disable-tensorflow-debugging-information)

**Comment by amahendrakar** [comment]
> Tensorflow warns of not having a GPU installed but you can work with CPU as well. To hide these errors and warnings reported by Tensorflow refer [here](https://stackoverflow.com/questions/35911252/disable-tensorflow-debugging-information)

@DavidMulvey2,
Could you please check @oke-aditya's comment and let us know if it helps. Thanks!

**Comment by DavidMulvey2** [comment]
Hi there

Thank you very much for looking into this but on the face of it, it doesn't seem to help because the reply seems to be about how to suppress error messages rather than resolving the underlying issue.

David
________________________________

**Comment by ymodak** [comment]
You may want to install tensorflow cpu package in this case.
`pip install tensorflow-cpu`

**Comment by DavidMulvey2** [comment]
That's a good tip thanks.

In reply to Aditya thanks for the question but in fact I couldn't do a fit which is why I had to raise the issue.

I have solved it myself temporarily by setting the GPU index to -1 but I would expect that Yasir's tip will provide a permanent solution if I need it.

So, I would be happy for you to close the issue now, most grateful for your assistance on this, stay safe everyone.

Thanks

David


________________________________

**Comment by oke-aditya** [comment]
CPU option is possible. Stay Safe and have a Happy Time working with Tensorflow. :+1:
